### PR TITLE
Always build a configdrive except for live ISO

### PR DESCRIPTION
**Pull request by dtantsur**
We have default meta data that we should pass in any case.

PR based on top of https://github.com/metal3-io/baremetal-operator/pull/900.

### CODE REVIEW

File: pkg/provisioner/ironic/ironic.go [hunk 1]
```diff
@@ -1239,6 +1239,59 @@ func (p *ironicProvisioner) Prepare(data provisioner.PrepareData, unprepared boo
 	return
 }
 
+func (p *ironicProvisioner) getConfigDrive(data provisioner.ProvisionData) (configDrive nodes.ConfigDrive, err error) {
+	// In theory, Ironic can support configdrive with live ISO by attaching
+	// it to another virtual media slot. However, Dell does not support two
+	// virtual media devices at the same time, so we shouldn't try it.
```
Comment (hardys) [inline pkg/provisioner/ironic/ironic.go:1245]: From recent discussions with @rhjanders I don't think Supermicro does either, so we should perhaps just say this is not reliably/commonly supported on all hardware types?

**Comment by hardys** [comment]
This lgtm - one small nit re the comment but IMO it makes sense to always pass the metal3 meta-data regardless of the user input in the spec.

I suspect that the reason this ended up originally conditional on the userData is that was at first the only field included in the config-drive, then others got added later (@zaneb do you recall if that's accurate perhaps?)

**Comment by zaneb** [comment]
> I suspect that the reason this ended up originally conditional on the userData is that was at first the only field included in the config-drive, then others got added later (@zaneb do you recall if that's accurate perhaps?)

Yes, I'm 99% sure it was unintentional for that reason. Metadata was added later in #448 and NetworkData in #348.

**Comment by zaneb** [comment]
Refamiliarising myself with the context... we bumped this out of #900 because we realised that always building a config drive could break the live-iso case. With the live-iso case carved out here, it should be safe to always build a config drive, which is probably the way it should always have worked, even though it is a user-visible change of behaviour.
/lgtm

**Pull request merged**